docs: follow-up to gateway migration in #687 (#695)
apollographql/router: docs/source/migrating-from-gateway.mdx

In the Apollo Router, logging is JSON-structured in production environments by default and the verbosity can be adjusted.  More advanced logging can be enabled through the use of Plugins (discussed later in this article).  For more information on the built-in logging, see our section on [Logging](./configuration/logging).

## Constructor options (including plugins) passed to `new ApolloServer({ ... })`

There are a large variety of uncommon parameters that can be passed to `new ApolloServer({ ... })`, though we'll focus on the most common ones:

- `context`
- `plugins`

To see the full list of options, see [`ApolloServer` options](https://www.apollographql.com/docs/apollo-server/api/apollo-server#options).  If you're using other options, additional migration steps may be necessary to replicate the same behavior.  Please open a discussion on our GitHub repository so we can better understand your needs and hopefully help you with a solution.

### `context`

The `context` parameter is an object that is which TODO

### `plugins`

> It was often necessary to write various plugins for Apollo Server, which is the foundation that Apollo Gateway runs on.  These plugins are passed to the `new ApolloServer({ ... })` constructor options in the `plugins` array.  Understanding what each plugin which was enabled will be important to note.

Reviewer: I can't parse: "Understanding what each plugin which was enabled will be important to note.". Should it be:
"It will be important to understand and document why each plugin was enabled."
?

Author: Agree.  And that's the sentiment, yes.  It should probably be more of an extension of the check-list idea I started further up.  How about?:

```suggestion
It was often necessary to write various plugins for Apollo Server, which is the foundation that Apollo Gateway runs on.  These plugins are passed to the `new ApolloServer({ ... })` constructor options in the `plugins` array.  First, take note of each plugin and its use-cases and add it to your check-list, then consider approaches to accomplish the same thing in the Apollo Router.  Plugins might not map one-to-one to plugins in the Apollo Router, but you'll find similar techniques..
```

> Generally speaking, Apollo Router is trying to minimize the number of cases where custom plugins are necessary to perform common behaviors, for example, setting custom HTTP status codes, transmitting telemetry data, etc.  To facilitate this, Apollo Router has a YAML-structured configuration based which enables common behaviors:

Reviewer: ```suggestion
Generally speaking, Apollo Router is trying to minimize the number of cases where custom plugins are necessary to perform common behaviors. For example, setting custom HTTP status codes, transmitting telemetry data, etc.  To facilitate this, Apollo Router has a YAML-structured configuration that enables common behaviors:
```

- Propagating particular/all headers to subgraphs
- OpenTelemetry instrumentation and distributed tracing support
- Adjusting logging verbosity

> To try to reduce the amount of user code that lives in each Apollo Router, prior to re-implementing existing Apollo Gateway plugin behavior in a new language (Rust), we encourage our community to start (or engage on existing) conversations about desired customizations in the [GitHub Discussions on the Router repository](https://github.com/apollographql/router/discussions/).  There are various core plugins we think the Apollo Router could offer out of the box and we'd like to help discuss their design.

Reviewer: ```suggestion
Prior to re-implementing existing Apollo Gateway plugin behavior in a new language (Rust), we encourage our community to start (or engage on existing) conversations about desired customizations in the [GitHub Discussions on the Router repository](https://github.com/apollographql/router/discussions/).  There are various core plugins we think the Apollo Router could offer out of the box and we'd like help to discuss their design.
```

Author: I was somewhat trying to paint the picture of some of the pains of shared user code living in a centralized place and what that means to evolving a federated graph (usually someone still has to "own" that code), but agree it's better without this.  That's probably deserving of a whole article. 😉
